packages/flutter/lib/src/widgets/text_editing_shortcuts.dart

```diff
@@ -0,0 +1,106 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/services.dart';
+
+import 'actions.dart';
+import 'framework.dart';
+import 'shortcuts.dart';
+import 'text_editing_intents.dart';
+
+/// A [Shortcuts] widget with the shortcuts used for the default text editing
+/// behavior.
+///
+/// This default behavior can be overridden by placing a [Shortcuts] widget
+/// lower in the widget tree than this. See [TextEditingActions] for an example
+/// of remapping a text editing [Intent] to a custom [Action].
+///
+/// {@tool snippet}
+///
+/// This example shows how to use an additional [Shortcuts] widget to override
+/// the left arrow key [Intent] and map it to the right arrow key instead.
+///
+/// If
+///
+/// ```dart
+/// final TextEditingController controller = TextEditingController(
+///   text: "Try using the keyboard's arrow keys and notice that left moves right.",
+/// );
+///
+/// @override
+/// Widget build(BuildContext context) {
+///   return Scaffold(
+///     body: Center(
+///       child: Shortcuts(
+///         shortcuts: <LogicalKeySet, Intent>{
+///           LogicalKeySet(LogicalKeyboardKey.arrowLeft): ArrowRightTextIntent(),
```

Comment:
re: my naming comment from above, would this be clearer as:

```suggestion
///           LogicalKeySet(LogicalKeyboardKey.arrowLeft): MoveCursorForwardTextIntent(),
```

Reply:
Yeah that seems to be right about RTL.  I've kept all naming related to left and right, so this one is: `MoveSelectionRightTextIntent`.